pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1037,6 +1049,12 @@ private void closeConsumerTasks() {
             possibleSendToDeadLetterTopicMessages.clear();
         }
 
+        if (!ackRequests.isEmpty()) {
+            ackRequests.forEach((key, value) -> value.callback
+                    .completeExceptionally(new TransactionConflictException("Consumer has closed!")));
```

Comment:
The TransactionConflictException does not make sense here. It's better to add a MessageAcknowledgeException.